test/control-plane/marshalling.go

```diff
@@ -0,0 +1,97 @@
+// SPDX-License-Identifier: Apache-2.0
+// Copyright Authors of Cilium
+
+package controlplane
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	k8sRuntime "k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/util/yaml"
+	"k8s.io/client-go/kubernetes/fake"
+
+	cilium_v2 "github.com/cilium/cilium/pkg/k8s/apis/cilium.io/v2"
+	slim_corev1 "github.com/cilium/cilium/pkg/k8s/slim/k8s/api/core/v1"
+	slim_fake "github.com/cilium/cilium/pkg/k8s/slim/k8s/client/clientset/versioned/fake"
+)
+
+//
+// Marshalling utilities for control-plane tests.
+//
+
+var (
+	// coreDecoder decodes objects using only the corev1 scheme
+	coreDecoder = newCoreSchemeDecoder()
+
+	// slimDecoder decodes objects with the slim scheme
+	slimDecoder = newSlimSchemeDecoder()
+
+	// ciliumDecoder decodes objects with the cilium v2 scheme
+	ciliumDecoder = newCiliumSchemeDecoder()
+)
+
+type schemeDecoder struct {
+	*k8sRuntime.Scheme
+}
+
+func (d schemeDecoder) unmarshal(in string) k8sRuntime.Object {
+	var obj unstructured.Unstructured
+	err := yaml.Unmarshal([]byte(in), &obj)
+	if err != nil {
+		panic(err)
```

Comment:
Just curious, why did you decide to have failures here panic? Think it would be good to have some context as to where this is being called from when panicing to help someone debugging a test.

Reply:
Same for all of these

Reply:
Mostly use of `panic` was because this was initially rather experimental, and as the stack trace with the error should've had enough context. But now that I look at this, it's much better that errors are bubbled up via `t.Fatalf` instead, especially if later on we'll be able to have multiple test cases in one binary. I'll clean all these up.